modules/light-clients/07-tendermint/types/update.go

```diff
@@ -60,6 +60,34 @@ func (cs ClientState) CheckHeaderAndUpdateState(
 		return nil, nil, err
 	}
 
+	// Check the earliest consensus state to see if it is expired, if so then set the prune height
+	// so that we can delete consensus state and all associated metadata.
+	var (
+		pruneHeight exported.Height
+		pruneError  error
+	)
+	pruneCb := func(height exported.Height) bool {
+		consState, err := GetConsensusState(clientStore, cdc, height)
+		if err != nil {
+			pruneError = err
+			return true
+		}
+		if consState.IsExpired(cs.TrustingPeriod, ctx.BlockTime()) {
+			pruneHeight = height
+		}
```

Comment:
I'd just use the existing client state `IsExpired` function, doesn't make sense to me to maintain two copies